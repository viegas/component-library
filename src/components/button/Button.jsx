import React from 'react';
import PropTypes from 'prop-types';

import * as Styled from './Button.styles';
import Loader from '../loader/Loader';

const Button = ({ type, disabled, busy, children, ...props }) => {
    const blockActions = disabled || busy;

    return (
        <Styled.Button {...props} type={blockActions ? 'disabled' : type}>
            {children}
            {busy && (
                <Styled.LoaderWrapper>
                    <Loader color="blue-700" />
                </Styled.LoaderWrapper>
            )}
        </Styled.Button>
    );
};

Button.propTypes = {
    type: PropTypes.oneOf(['primary', 'secondary']),
    disabled: PropTypes.bool,
    busy: PropTypes.bool,
    children: PropTypes.node,
};

Button.defaultProps = {
    type: 'primary',
    disabled: false,
    busy: false,
    children: null,
};

export default Button;
