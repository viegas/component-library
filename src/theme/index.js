const theme = {
    spacings: {
        small: '5px',
        medium: '10px',
        large: '20px',
    },
    font: {
        size: {
            info: '12px',
            text: '16px',
            subTitle: '18px',
            title: '24px',
        },
        weight: {
            ligth: 300,
            regular: 400,
            semibold: 600,
            bold: 700,
            extraBold: 800,
        },
    },
    colors: {
        neutral: '#bbbbbb',
        'light-neutral': '#e4e4e4',
        'dark-red': '#e7040f',
        red: '#ff4136',
        'light-red': '#ff725c',
        orange: '#ff6300',
        gold: '#ffb700',
        yellow: '#ffde37',
        'light-yellow': '#fbf1a9',
        purple: '#5e2ca5',
        'light-purple': '#a463f2',
        'dark-pink': '#d5008f',
        'hot-pink': '#ff41b4',
        pink: '#ff80cc',
        'light-pink': '#ffa3d7',
        'dark-green': '#137752',
        green: '#19a974',
        'light-green': '#9eebcf',
        navy: '#001b44',
        'dark-blue': '#00449e',
        blue: '#357edd',
        'light-blue': '#96ccff',
        'lightest-blue': '#cdecff',
        'washed-blue': '#f6fffe',
        'washed-green': '#e8fdf5',
        'washed-yellow': '#fffceb',
        'washed-red': '#ffdfdf',
        white: '#ffffff',
        black: '#000000',
        transparent: '#ffffff00',
    },
    media: {
        breakpoints: {
            md: 768,
            lg: 1024,
        },
    },
};

export const mediaQueriesGenerator = (breakpoints) =>
    Object.keys(breakpoints).reduce((acc, breakpoint) => {
        const size = theme.media.breakpoints[breakpoint];
        const mediaQuery = `(min-width: ${size}px)`;

        acc[breakpoint] = `@media ${mediaQuery}`;
        return acc;
    }, {});

theme.media.query = mediaQueriesGenerator(theme.media.breakpoints);

export default theme;
