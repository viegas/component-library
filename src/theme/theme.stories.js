import React from 'react';
import { storiesOf } from '@storybook/react';

import theme from './index';

const boxStyle = {
    padding: '20px',
    paddingTop: '0px',
    marginBottom: '20px',
    borderBottom: '1px solid gray',
};

const Theme = () => {
    return (
        <div>
            <div style={boxStyle}>
                <p>Spacings</p>
                <div style={{ backgroundColor: 'gray', overflow: 'hidden' }}>
                    {Object.keys(theme.spacings).map((key) => (
                        <div
                            style={{
                                backgroundColor: 'white',
                                marginLeft: theme.spacings[key],
                                marginBottom: theme.spacings[key]
                            }}
                        >
                            {key} - {theme.spacings[key]}
                        </div>
                    ))}
                </div>
            </div>
            <div style={boxStyle}>
                <p>Colors</p>
                <table>
                    <tbody>
                        {Object.keys(theme.colors).map((key) => (
                            <tr>
                                <td
                                    style={{
                                        background: theme.colors[key],
                                        padding: '20px',
                                        width: '33%',
                                    }}
                                />
                                <td
                                    style={{
                                        color: theme.colors[key],
                                        padding: '20px',
                                    }}
                                >
                                    Aa
                                </td>
                                <td
                                    style={{
                                        padding: '20px',
                                    }}
                                >
                                    {key} - {theme.colors[key]}
                                </td>
                            </tr>
                        ))}
                    </tbody>
                </table>
            </div>
            <div style={boxStyle}>
                <p>Font Size</p>
                {Object.keys(theme.font.size).map((key) => (
                    <div style={{ fontSize: theme.font.size[key] }}>
                        {key} - {theme.font.size[key]}
                    </div>
                ))}
            </div>
            <div style={boxStyle}>
                <p>Font Weights</p>
                {Object.keys(theme.font.weight).map((key) => (
                    <div style={{ fontWeight: theme.font.weight[key] }}>
                        {key} - {theme.font.weight[key]}
                    </div>
                ))}
            </div>
        </div>
    );
};

storiesOf('Styles', module).add('theme.js', () => {
    return <Theme />;
});
