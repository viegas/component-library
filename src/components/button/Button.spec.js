import Button from './Button';
import * as Styled from './Button.styles';
import theme from '../../../theme';

describe('<Button />', () => {
    it('Should render correctly', () => {
        const wrapper = shallow(<Button>Test</Button>);

        expect(wrapper).toMatchSnapshot();
    });

    styledComponentsSnapshots(Styled, ['Icon', 'BusyLoader']);
});
