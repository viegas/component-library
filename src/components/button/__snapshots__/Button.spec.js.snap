// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> <Styled.BusyLoader /> Should render correctly 1`] = `
.c0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c0 > svg {
  display: block;
}

<span
  className="c0"
/>
`;

exports[`<Button /> <Styled.Button /> Should render correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  max-width: 320px;
  border-radius: 9999px;
  padding: 0 20px;
  font-size: 12px;
  text-align: center;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border: solid 1px;
  background-color: #FFFFFF;
  border-color: #cfcfcf;
  color: #1A8099;
}

.c0 > span {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  border-color: #3EBEDE;
  color: #3EBEDE;
}

<button
  className="c0"
  color="secondary"
>
  Test
</button>
`;

exports[`<Button /> <Styled.Button /> Should render disabled correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  max-width: 320px;
  border-radius: 9999px;
  padding: 0 20px;
  font-size: 12px;
  text-align: center;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border: solid 1px;
  background-color: #FFFFFF;
  border-color: #bbbbbb;
  color: #bbbbbb;
  cursor: not-allowed;
}

.c0 > span {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  className="c0"
  color="secondary"
  disabled={true}
>
  Test
</button>
`;

exports[`<Button /> <Styled.Button /> Should render primary color correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  max-width: 320px;
  border-radius: 9999px;
  padding: 0 20px;
  font-size: 12px;
  text-align: center;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border: none;
  background-color: #1A8099;
  color: #FFFFFF;
}

.c0 > span {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  background-color: #3EBEDE;
}

<button
  className="c0"
  color="primary"
>
  Test
</button>
`;

exports[`<Button /> <Styled.Button /> Should render primary disabled correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  max-width: 320px;
  border-radius: 9999px;
  padding: 0 20px;
  font-size: 12px;
  text-align: center;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border: none;
  background-color: #bbbbbb;
  color: #6a6a6a;
  cursor: not-allowed;
}

.c0 > span {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  className="c0"
  color="primary"
  disabled={true}
>
  Test
</button>
`;

exports[`<Button /> <Styled.Button /> Should render selected correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  max-width: 320px;
  border-radius: 9999px;
  padding: 0 20px;
  font-size: 12px;
  text-align: center;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  background-color: #0E5061;
  color: #FFFFFF;
  border: none;
}

.c0 > span {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  className="c0"
  color="secondary"
  selected={true}
>
  Test
</button>
`;

exports[`<Button /> <Styled.Icon /> Should render correctly 1`] = `
.c0 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 5px;
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c0 > svg {
  display: block;
}

<span
  className="c0"
/>
`;

exports[`<Button /> Should render busy correctly 1`] = `
<Buttonstyles__Button
  as="button"
  color="secondary"
  disabled={true}
  selected={false}
  type="button"
>
  <span>
    Test
    <Buttonstyles__BusyLoader>
      <BusyLoader
        timer={10}
      />
    </Buttonstyles__BusyLoader>
  </span>
</Buttonstyles__Button>
`;

exports[`<Button /> Should render correctly 1`] = `
<Buttonstyles__Button
  as="button"
  color="secondary"
  disabled={false}
  selected={false}
  type="button"
>
  <span>
    Test
  </span>
</Buttonstyles__Button>
`;

exports[`<Button /> Should render like an anchor correctly 1`] = `
<Buttonstyles__Button
  as="a"
  color="secondary"
  disabled={false}
  selected={false}
>
  <span>
    Test
  </span>
</Buttonstyles__Button>
`;

exports[`<Button /> Should render like an submit correctly 1`] = `
<Buttonstyles__Button
  as="button"
  color="secondary"
  disabled={false}
  selected={false}
  type="submit"
>
  <span>
    Test
  </span>
</Buttonstyles__Button>
`;

exports[`<Button /> Should render with icon correctly 1`] = `
<Buttonstyles__Button
  as="button"
  color="secondary"
  disabled={false}
  selected={false}
  type="button"
>
  <span>
    <Buttonstyles__Icon>
      <Icon
        size="12px"
        type="Calendar"
      />
    </Buttonstyles__Icon>
    Test
  </span>
</Buttonstyles__Button>
`;
